corebehrt/classes/values.py

```diff
@@ -0,0 +1,114 @@
+import dask.dataframe as dd
+from corebehrt.common.config import instantiate_function
+
+
+class ValueCreator:
+    """
+    A class to create features from patient information and concepts DataFrames.
+    We create background, death, age, absolute position, and segments features.
+    Parameters
+    ----------
+    origin_point : datetime or dict, optional
+        The origin point for calculating absolute positions. If a dict is provided, it's used to create a datetime object.
+    background_vars : list, optional
+        List of background variable names to include.
+
+    Examples
+    --------
+    >>> feature_creator = FeatureCreator(origin_point=datetime(2020, 1, 26), background_vars=["GENDER", "AGE"])
+    >>> features = feature_creator(patients_info, concepts)
+    """
+
+    def __init__(
+        self,
+        value_type: str,
+        value_type_kwargs: dict = None,
+        normalize_args: dict = None,
+    ):
+        self.value_type = value_type  # values.get("value_type", None)
+        self.value_type_kwargs = (
+            value_type_kwargs  # values.get("value_type_kwargs", None)
+        )
+        self.normalize_args = normalize_args  # values.get("normalize_args", None)
+        if self.normalize_args is not None:
+            self.normalize_args["func"] = instantiate_function(
+                self.normalize_args["func"]
+            )
+
+    def __call__(
+        self,
+        concepts: dd.DataFrame,
+    ) -> dd.DataFrame:
+        if self.value_type == "binned":
+            return self.add_binned_values(
+                concepts,
+                self.value_type_kwargs["multiplication_factor"],
```

Comment:
This will fail if self.value_type_kwargs is `None`, as is the default value? So either we add a `.get` or we remove the default `None` in the `__init__`